types/types.go

```diff
@@ -441,7 +441,7 @@ type LoggingConfig struct {
 // DeployConfig the deployment configuration for a service
 type DeployConfig struct {
 	Mode           string         `yaml:",omitempty" json:"mode,omitempty"`
-	Replicas       *uint64        `yaml:",omitempty" json:"replicas,omitempty"`
+	Replicas       *uint64        `yaml:"" json:"replicas"`
```

Comment:
Is this change intentional? Other pointer fields in this struct do have `omitempty`.
I’m not sure what gets serialized in `Replicas` is `nil`.
Or maybe we’re only loading, so it doesn’t really matter?

Reply:
yes it is, so that we get value serialized by `compose config` when set to `0`.
a `nil` value is never serialized.

Reply:
But `omitempty` here is for the pointer, if it’s non-`nil` and points to a `0` value, it will be serialized even if the field has `omitempty`.
Even if it has the same behaviour with or without `omitempty`, I think if only for consistency sake with the rest of the pointers, it should probably have an `omitempty`?

Reply:
`omitempty` does not apply to the pointer but to the actual value. nil pointer never get serialized.
Here we WANT `0` to be serialized (default value we would like to ignore is `1`)

Reply:
It’s the zero value of the field that matters (here it’s a pointer, so the zero value is `nil`): https://pkg.go.dev/gopkg.in/yaml.v2#Marshal (it’s v2 which I think compose-go is using, but it’s the same in v3):

> omitempty:    Only include the field if it's not set to the zero
             value for the type or to empty slices or maps.

`nil` pointers without `omitempty` yields a `null` value in the output: https://play.golang.org/p/Hk2qfCq9ccT
With `omitempty` the `nil` pointer gets omitted: https://play.golang.org/p/EVjf0jL2uN4
but not if the pointer is non-nil, even if it points to a 0 value: https://play.golang.org/p/7dype4Fh4cP

Reply:
indeed. I was confused as `deploy` is istelf a pointer so was omitted.

Reply:
so `replicas` is always serialized when non nil, and we only set value from `scale` by normalization if value != 1